.github/workflows/nodejs.yml

```diff
@@ -41,4 +41,9 @@ jobs:
           npm test
         env:
           CI: true
+      - name: check formatting
```

Comment:
Should we just check here or should we actually fix it? I don't have an answer just wondering

Reply:
My intent for the CI step is that it should be more or less an optional indicator that reviewers can quickly look at to diagnose if a PR has any formatting issues. Generally speaking, I don't think we would want the CI itself to apply formatting, since we would have to amend the commit with the correctly formatted code to the PR in some way, which would be quite surprising for developers. This would also make it almost impossible for us to skip any rare edge cases where formatting with Prettier causes a file to get mangled, since skipping Prettier locally would cause the CI to format the code anyway (this being part of the reasoning for making the CI check optional and non-blocking).